### PR TITLE
Fixes IAM Permissions

### LINKED ISSUE #4491: Document Minimum AWS IAM Roles for DynamoDB/S3
## Summary
The AWS IAM roles that we recommend, in documentation for DynamoDB, permit all DynamoDB actions. Additionally, for S3 there are a few additional actions that are required. 
```json
{
   "Sid": "AllAPIActionsOnTeleportClusterStateStreams",
  "Effect": "Allow",
  "Action": "dynamodb:*",
  "Resource": "arn:aws:dynamodb:REGION:ACCOUNTID:table/DYNAMODB-RESOURCE-A/stream/*"
}
```

https://gravitational.com/teleport/docs/aws-oss-guide/#dynamodb
https://gravitational.com/teleport/docs/aws-oss-guide/#iam

### Audience
AWS Users

### Location
Update Existing Docs

### PR DESCRIPTION
**Pull request by benarent**
Fixes https://github.com/gravitational/teleport/issues/4491

### CODE REVIEW

File: docs/4.4/aws-oss-guide.md [hunk 1]
```diff
@@ -185,42 +190,65 @@ An example policy is shown below:
...
+                "dynamodb:UpdateItem",
+                "dynamodb:DescribeTimeToLive",
+                "dynamodb:UpdateTable"
+            ],
+            "Resource": [
+                "arn:aws:dynamodb:<region>:<account_id>:table/<events_table_name>",
+                "arn:aws:dynamodb:<region>:<account_id>:table/<events_table_name>/index/*"
+            ]
         },
         {
```
Comment (webvictim) [inline docs/4.4/aws-oss-guide.md:239]: I _think_ that we only use a locks table in our reference Terraform - we only need `cluster_state_table_name` and `events_table_name` for regular Teleport setups. You can delete this third section.

Comment (benarent) [reply docs/4.4/aws-oss-guide.md:239]: @travelton Do you remember why we had these locks, I'm pretty sure I've not had to do it myself.

Comment (webvictim) [reply docs/4.4/aws-oss-guide.md:239]: I'm pretty sure the only use of the lock table is here: https://github.com/gravitational/teleport/blob/11a82524bdb7a6a7059955818ccb97f1ea407e5d/assets/aws/files/bin/teleport-lock#L25

We can just remove this.

Comment (travelton) [reply docs/4.4/aws-oss-guide.md:239]: Agreed. I believe this IAM role is specific to the HA Terraform example.

Comment (benarent) [reply docs/4.4/aws-oss-guide.md]: Fixed.

**Review by webvictim — APPROVED** [review]

**Pull request merged**